Modules/FOCAL/src/PedestalCalibTask.cxx

```diff
@@ -0,0 +1,135 @@
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+///
+/// \file   PedestalCalibTask.cxx
+/// \author My Name
+///
+
+#include <TCanvas.h>
+#include <TH1.h>
+#include <TH2.h>
+
+#include "FOCALCalib/PadPedestal.h"
+
+#include "QualityControl/QcInfoLogger.h"
+#include "FOCAL/PedestalCalibTask.h"
+#include <Framework/InputRecord.h>
+#include <Framework/InputRecordWalker.h>
+#include <Framework/DataRefUtils.h>
+
+namespace o2::quality_control_modules::focal
+{
+
+PedestalCalibTask::PedestalCalibTask()
+{
+  std::fill(mPedestalChannel.begin(), mPedestalChannel.end(), nullptr);
+  std::fill(mPedestalPosition.begin(), mPedestalPosition.end(), nullptr);
+}
+
+PedestalCalibTask::~PedestalCalibTask()
+{
+  for (auto& hist : mPedestalChannel) {
+    delete hist;
+  }
+  for (auto& hist : mPedestalPosition) {
+    delete hist;
+  }
+}
+
+void PedestalCalibTask::initialize(o2::framework::InitContext& /*ctx*/)
+{
+  for (auto layer = 0; layer < mPedestalChannel.size(); layer++) {
+    mPedestalChannel[layer] = new TH1D(Form("mPedestalChannelLayer%d", layer), Form("Pedestals in layer %d", layer), o2::focal::constants::PADLAYER_MODULE_NCHANNELS, -0.5, o2::focal::constants::PADLAYER_MODULE_NCHANNELS - 0.5);
+    mPedestalChannel[layer]->SetXTitle("Channel ID");
+    mPedestalChannel[layer]->SetYTitle("Pedestal (ADC counts)");
+    getObjectsManager()->startPublishing(mPedestalChannel[layer]);
+    mPedestalPosition[layer] = new TH2D(Form("mPedestalPositionLayer%d", layer), Form("Pedestals in layer %d", layer), o2::focal::PadMapper::NCOLUMN, -0.5, o2::focal::PadMapper::NCOLUMN - 0.5, o2::focal::PadMapper::NROW, -0.5, o2::focal::PadMapper::NROW - 0.5);
+    mPedestalPosition[layer]->SetXTitle("Column");
+    mPedestalPosition[layer]->SetYTitle("Row");
+    getObjectsManager()->startPublishing(mPedestalPosition[layer]);
+  }
+}
+
+void PedestalCalibTask::startOfActivity(Activity& activity)
+{
+  ILOG(Debug, Devel) << "startOfActivity " << activity.mId << ENDM;
+  reset();
+}
+
+void PedestalCalibTask::startOfCycle()
+{
+  ILOG(Debug, Devel) << "startOfCycle" << ENDM;
+}
+
+void PedestalCalibTask::monitorData(o2::framework::ProcessingContext& ctx)
+{
+  ILOG(Debug, Support) << "Start monitoring data" << ENDM;
+  bool hasPedestalsCCDB = false;
+  for (auto&& input : o2::framework::InputRecordWalker(ctx.inputs())) {
+    // get message header
+    if (input.header != nullptr && input.payload != nullptr) {
+      auto payloadSize = o2::framework::DataRefUtils::getPayloadSize(input);
+      const auto* header = o2::framework::DataRefUtils::getHeader<header::DataHeader*>(input);
+      if (payloadSize) {
+        if ((strcmp(header->dataOrigin.str, "CLP") == 0) && (strcmp(header->dataDescription.str, "FOC_PADPEDESTALSCLP") == 0)) {
+          ILOG(Info, Support) << "Pedestals for Pads found" << ENDM;
```

Comment:
I am always fearful of logging within `monitorData`. It can easily create floods of logs as this method is called at high frequency. Perhaps you could move it to `Debug` or make it so that it does not log more than X per minute: 
```
  static AliceO2::InfoLogger::InfoLogger::AutoMuteToken msgLimit(LogInfoSupport, 1, 600); // send it only every 10 minutes
    ILOG_INST.log(msgLimit, "blabla");
```

Reply:
I can move later the log level to debug but as this is a QC component only attached to a pedestal run, which monitors the object stored in the CCDB, the monitorData will be called only once, when the CCDB object is published. So there will be no message flood. For the time being while we are still developing the pedestal run I would like to keep the logging at level info. Once the development is done I will decrease the level to debug.